feat: webhook and schedule triggers

a7m-1st opened the pull request:
## Description
### Trigger System Core Implementation
- **WebhookTrigger**: Complete HTTP webhook server implementation with aiohttp backend
  - Configurable port, path, and host binding
  - Support for JSON and form-encoded payloads  
  - Comprehensive error handling and logging
  - Integration with TriggerManager for event processing

- **TriggerManager**: Central orchestration system for all trigger types
  - Supports ChatAgent and Workforce integration patterns
  - Event-to-Task conversion for seamless workflow integration
  - Automatic trigger registration and lifecycle management
  - Database adapter support for persistent trigger state

### Slack Integration & Authentication
- **SlackAuth**: Production-ready webhook signature verification
  - HMAC-SHA256 signature validation following Slack security best practices
  - Timestamp freshness checks to prevent replay attacks
  - URL verification challenge/response handling for Slack App setup
  - Environment variable configuration (SLACK_SIGNING_SECRET, SLACK_BOT_TOKEN)
  
- **Slack Webhook Example**: Complete integration demonstration
  - Event type handling (url_verification, event_callback, messages, app_mention)
  - Proper error handling and graceful degradation without credentials
  - Real-world usage patterns with comprehensive logging

### TODOs & Next Steps
- [x] Add unit tests for trigger system components with appropriate pytest markers
- [x] Add configuration validation and better error messages
- [ ] Implement trigger scheduling and rate limiting features
- [ ] Implement specific DatabaseTrigger adapters (PostgreSQL)
- [ ] Implement trigger state persistence and recovery mechanisms

### Technical Notes
- All components follow CAMEL's type annotation requirements
- Proper logger usage instead of print statements throughout
- Environment variable configuration for secure credential management
- Comprehensive error handling with meaningful context messages

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have linked this PR to an issue using the Development section on the right sidebar or by adding `Fixes #issue-number` in the PR description (**required**)
- [x] I have checked if any dependencies need to be added or updated in `pyproject.toml` and `uv lock`
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*)
- [x] I have updated the documentation if needed:
- [x] I have added examples if this is a new feature

If you are unsure about any of these, don't hesitate to ask. We are here to help!

LuoPengcheng12138 reviewed (approved):
All looks good to me!

a7m-1st commented on camel/triggers/trigger_manager.py:
```diff
@@ -0,0 +1,278 @@
+                and self.chat_agent
+            ):
+                # Use provided prompt or convert event to prompt
+                if self.default_prompt:
```
Oh didn't realize yes you are right @LuoPengcheng12138 , making a fix asap !

a7m-1st commented:
Hi there @LuoPengcheng12138 I have added some small updates:
1. According to your comment, simplified the payload passing (passing the whole payload now)
2. Added a new allow_duplicate_events param bcz turns out I receive duplicate events sometimes

fengju0213 commented on camel/auth/slack_auth.py:
```diff
@@ -0,0 +1,262 @@
+            logger.warning(
+                "No Slack signing secret configured, skipping verification"
+            )
+            return True
```
should be False？

a7m-1st replied:
It was actually optional, but ya you are right.
Just fixed If user called the function without providing signing_secret it will return `AuthenticationError` ✅

lightaime commented on camel/triggers/trigger_manager.py:
```diff
@@ -0,0 +1,332 @@
+    """Enum defining types of callback handlers for trigger events"""
+
+    WORKFORCE = "workforce"
+    CHATAGENT = "chatagent"
```
```suggestion
    CHAT_AGENT = "chat_agent"
```

a7m-1st commented on camel/triggers/trigger_manager.py:
```diff
@@ -0,0 +1,332 @@
+                    )
+
+                # Process with ChatAgent
+                response = self.chat_agent.step(prompt)
```
All right using `astep` I hope that is okay? 
I will refactor it with the `TriggerEventHandler` as well
